### ISSUE TITLE
Progress handler always returns 0 on iOS

### ISSUE THREAD

**Issue by dasantonym** [AlwaysLoveme/capacitor-plugin-filedownload#10]
The progress handler gets called, but I only get the value `0`.

Plugin version: 1.0.8
Capacitor version: 4.8.0
iOS Target: 13.0

**Comment by AlwaysLoveme** [comment]
> The progress handler gets called, but I only get the value `0`.
> 
> Plugin version: 1.0.8 Capacitor version: 4.8.0 iOS Target: 13.0

is it downloaded successfully?

**Comment by dasantonym** [comment]
Yes, downloading works, but each progress event only emits a zero.

**Comment by dasantonym** [comment]
I figured it out, it's not iOS-specific. If the web server does not supply a `Content-Length` header, then the calculation for progress does not work.

Not sure how to work around this. Maybe add a property named `bytesReceived` that just holds the bytes in case the total length is unknown?
